src/main/java/org/elasticsearch/index/engine/InternalEngineFactory.java

```diff
@@ -18,9 +18,54 @@
  */
 package org.elasticsearch.index.engine;
 
+import org.elasticsearch.cluster.metadata.IndexMetaData;
+import org.elasticsearch.common.util.concurrent.ReleasableLock;
+
 public class InternalEngineFactory implements EngineFactory {
     @Override
     public Engine newReadWriteEngine(EngineConfig config, boolean skipTranslogRecovery) {
+        // On a shared filesystem, we need to handle recovery slightly
```

Comment:
can we move this to it's own class? and then use the same pattern as all the other Mockengine classes?